extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -48,19 +51,31 @@
   private static final Logger LOG = new Logger(OpenCensusProtobufInputRowParser.class);
 
   private static final String SEPARATOR = "-";
-  public static final String NAME = "name";
-  public static final String VALUE = "value";
-  public static final String TIMESTAMP_COLUMN = "timestamp";
+  private static final String DEFAULT_METRIC_DIMENSION = "name";
+  private static final String VALUE = "value";
+  private static final String TIMESTAMP_COLUMN = "timestamp";
+  private static final String DEFAULT_RESOURCE_PREFIX = "resource.";
   private final ParseSpec parseSpec;
   private final List<String> dimensions;
 
+  private final String metricDimension;
+  private final String metricLabelPrefix;
+  private final String resourceLabelPrefix;
+
   @JsonCreator
   public OpenCensusProtobufInputRowParser(
-      @JsonProperty("parseSpec") ParseSpec parseSpec
+      @JsonProperty("parseSpec") ParseSpec parseSpec,
+      @JsonProperty("metricDimension") String metricDimension,
+      @JsonProperty("metricLabelPrefix") String metricPrefix,
+      @JsonProperty("resourceLabelPrefix") String resourcePrefix
   )
   {
     this.parseSpec = parseSpec;
     this.dimensions = parseSpec.getDimensionsSpec().getDimensionNames();
+    this.metricDimension = !Strings.isNullOrEmpty(metricDimension) ? metricDimension : DEFAULT_METRIC_DIMENSION;
+    this.metricLabelPrefix = NullHandling.nullToEmptyIfNeeded(metricPrefix);
```

Comment:
why would we be using `NullHandling` here instead of  `Strings.nullToEmpty()`? `NullHandling` is only needed to manipulate actual values, but not for dimensions names.

Reply:
Druid code checks doesn't allow to use `Strings.nullToEmpty()`.
https://github.com/confluentinc/druid/blob/d67772c02ce230c3be339d8a1ce336405d89381c/codestyle/checkstyle.xml#L205

Reply:
that's to avoid possible bugs with null handling in values, you can override that check in this particular case, or use `StringUtils.nullToEmptyNonDruidDataString` from `org.apache.druid.java.util.common.StringUtils`

Reply:
Done.